File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IVideosInCatalogView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.Video;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;

public interface IVideosInCatalogView extends IMvpView, IErrorView, IAccountDependencyView {
    void displayList(List<Video> videos);

    void notifyListChanged();

    void displayRefreshing(boolean refresing);
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/IVideosInCatalogView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.Video
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface IVideosInCatalogView : IMvpView, IErrorView, IAccountDependencyView {
    fun displayList(videos: List<Video>)
    fun notifyListChanged()
    fun displayRefreshing(refresing: Boolean)
}